blazorApp/Program.cs

```diff
@@ -1,11 +1,27 @@
 using Microsoft.AspNetCore.Components.Web;
 using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
 using blazorApp;
+using blazorApp.Services;
 
 var builder = WebAssemblyHostBuilder.CreateDefault(args);
 builder.RootComponents.Add<App>("#app");
 builder.RootComponents.Add<HeadOutlet>("head::after");
 
-builder.Services.AddScoped(sp => new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) });
+builder.Services.AddSingleton<VsCodeHelper>();
+builder.Services.AddScoped<HttpClient>(sp =>
+{
+    var vsCodeHelper = sp.GetRequiredService<VsCodeHelper>();
+    var localResourceRoot = vsCodeHelper.AsWebViewResourceFileUri(relativeUri: "");
+    return new()
+    {
+        BaseAddress = new(localResourceRoot),
+    };
+});
 
-await builder.Build().RunAsync();
+var host = builder.Build();
+
+// Perform VSCode-specific setup before running the host.
+var vsCodeHelper = host.Services.GetRequiredService<VsCodeHelper>();
+vsCodeHelper.Initialize();

```

Comment:
are there cases when this initialize won't be loaded before line 13 runs?

Reply:
We should be good here since `Initialize()` completes synchronously. Line 13 only runs when an `HttpClient` gets requested from the DI container, which won't have happened at this point in time.